### PR TITLE
Update protovalidate, connect-go, and other go deps

### PR DESCRIPTION
**Pull request by pkwarren**

### CODE REVIEW

File: private/bufpkg/bufcheck/buflint/internal/buflintvalidate/field.go [hunk 1]
```diff
@@ -258,7 +258,7 @@ func checkFieldFlags(
 			adder.getFieldRuleName(),
 		)
 	}
-	if fieldConstraints.GetRequired() && fieldConstraints.GetIgnoreEmpty() {
+	if fieldConstraints.GetRequired() && (fieldConstraints.GetIgnoreEmpty() || fieldConstraints.GetIgnore() == validate.Ignore_IGNORE_EMPTY) {
```
Comment (pkwarren) [inline private/bufpkg/bufcheck/buflint/internal/buflintvalidate/field.go:261]: Kept using the previous deprecated field and also added the new enum with the same semantics. We can remove this when protovalidate reaches v1 and the previous ignore empty boolean is removed. See https://github.com/bufbuild/protovalidate/pull/137 for more details.

cc @rodaine

Comment (bufdev) [reply private/bufpkg/bufcheck/buflint/internal/buflintvalidate/field.go:261]: https://github.com/bufbuild/registry-proto/pull/75 means we probably need to update this further to use the non-deprecated name

File: .golangci.yml [hunk 2]
```diff
@@ -186,6 +186,9 @@ issues:
     - linters:
         - staticcheck
       text: "GetDeprecatedLegacyJsonFieldConflicts is deprecated"
+    - linters:
+        - staticcheck
+      text: "GetIgnoreEmpty is deprecated"
```
Comment (bufdev) [inline .golangci.yml:191]: Would optimally scope this to the specific file (the one above should be too, but doesn't need to be in this PR)

Comment (pkwarren) [reply .golangci.yml:191]: Fixed.

File: private/bufpkg/bufcheck/buflint/internal/buflintvalidate/field.go (hunk 1 above)
Comment (pkwarren) [reply private/bufpkg/bufcheck/buflint/internal/buflintvalidate/field.go]: Updated to the latest protovalidate.

**Review by rodaine — APPROVED** [review]

**Review by bufdev — APPROVED** [review]

**Pull request merged**